src/command_modules/azure-cli-billing/azure/cli/command_modules/billing/custom.py

```diff
@@ -0,0 +1,20 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def cli_billing_list_invoices(client, generate_url=False):
+    '''List all available invoices of the subscription'''
+    invoices = client.list(expand='downloadUrl' if generate_url else None)
+    return list(invoices)
+
+
+def cli_billing_get_invoice(client, name):
+    '''Retrieve invoice of specific name of the subscription'''
+    return client.get(name)
```

Comment:
From what you have here I would recommend removing this and have "invoice show" be via reflection.  *However*... you have an "invoice show" and an "invoice show-latest" command. This actually looks like the ideal opportunity to fold these into a single "invoice show" command where `name=None` makes name optional.  It would look something like:

```Python
def cli_billing_get_invoice(client, name=None):
  if name:
    return client.get(name)
  else:
    return client.get_latest(name)
```

This is very similar to commands throughout the CLI for retrieving things for the entire subscription or within a resource group.  Instead of two commands (because they are two SDK methods) there's a single command that chooses the right call based on whether resource group is specified.